### ISSUE TITLE
Flaky-test: ProxyConnectionThrottlingTest.testInboundConnection

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#17665]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/lhotari/pulsar/actions/runs/3058452434/jobs/4934839318#step:10:433

### Exception stacktrace

```
  Error:  Tests run: 4, Failures: 1, Errors: 0, Skipped: 3, Time elapsed: 18.584 s <<< FAILURE! - in org.apache.pulsar.proxy.server.ProxyConnectionThrottlingTest
  Error:  testInboundConnection(org.apache.pulsar.proxy.server.ProxyConnectionThrottlingTest)  Time elapsed: 2.474 s  <<< FAILURE!
  java.lang.AssertionError: expected [4] but found [5]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.proxy.server.ProxyConnectionThrottlingTest.testInboundConnection(ProxyConnectionThrottlingTest.java:111)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
test was added in #17167 . @mattisonchao would you mind fixing this?

**Comment by mattisonchao** [comment]
Sure

**Comment by poorbarcode** [comment]
Can this issue be closed?

**Comment by mattisonchao** [comment]
fixed by #17883
